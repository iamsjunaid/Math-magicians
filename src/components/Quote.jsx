import React, { useState, useEffect } from 'react';
import './styles/quote.css';

export default function Quote() {
  const [data, setData] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=computers',
          {
            headers: {
              'X-Api-Key': '8xq3zwv2C1lnUvRdoSwiig==qMdjeZk1bIzxlyTQ',
            },
          },
        );
        const json = await res.json();
        setData(json);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [setData, setIsLoading]);

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (hasError) {
    return <div>Something went wrong</div>;
  }
  return (
    <>
      {data.map((item) => (
        <div key={item} className="quote-content content">
          <p key={item.quote} className="quote">
            {item.quote}
          </p>
          <br />
          <p key={item.author} className="author">
            -
            {item.author}
          </p>
        </div>
      ))}
    </>
  );
}
