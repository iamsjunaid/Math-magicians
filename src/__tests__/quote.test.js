import { render } from '@testing-library/react';
import Quote from '../components/Quote';

test('renders FetchQuote correctly', () => {
  const { container } = render(
    <Quote />,
  );
  expect(container).toMatchSnapshot();
});
