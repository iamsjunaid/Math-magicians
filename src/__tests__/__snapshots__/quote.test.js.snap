// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FetchQuote correctly 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
